### PR TITLE
[ISSUE #9943] Resolve issue Config cas update not work

### PR DESCRIPTION
**Pull request by karsonto**
Resolve Issue #9943

Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: config/src/main/java/com/alibaba/nacos/config/server/service/repository/embedded/BaseDatabaseOperate.java [hunk 1]
```diff
@@ -213,18 +214,30 @@ default Boolean update(TransactionTemplate transactionTemplate, JdbcTemplate jdb
...
+                    listResult.add(update > 0 || !necessary);
                 });
                 if (consumer != null) {
-                    consumer.accept(Boolean.TRUE, null);
+                    if (!listResult.stream().anyMatch(e -> !e)) {
+                        consumer.accept(Boolean.TRUE, null);
+                        return Boolean.TRUE;
+                    }
+                    consumer.accept(Boolean.FALSE, null);
+                    return Boolean.FALSE;
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/service/repository/embedded/BaseDatabaseOperate.java:234]: if one operation fails ，the prev operations should rollback in a transaction context, only returning false may not rollback prev operations.

File: config/src/main/java/com/alibaba/nacos/config/server/service/sql/EmbeddedStorageContextUtils.java [hunk 2]
```diff
@@ -46,12 +46,13 @@ public class EmbeddedStorageContextUtils {
      * @param sql  sql
      * @param args argument list
      */
-    public static void addSqlContext(String sql, Object... args) {
+    public static void addSqlContext(String sql, boolean necessary, Object... args) {
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/service/sql/EmbeddedStorageContextUtils.java:49]: I think creating a new overload method is a better way ,and  many caller methods do not need modify

Comment (shiyiyue1102) [reply config/src/main/java/com/alibaba/nacos/config/server/service/sql/EmbeddedStorageContextUtils.java:49]: besides，the meaning of 'necessary' is not very clear for me, would you please explain it？
I think it is a transaction meaning, if this operation fails ,all operation should rollback, if so，can we change to a more accurate param name or add a detailed commentary  .

Comment (karsonto) [reply config/src/main/java/com/alibaba/nacos/config/server/service/sql/EmbeddedStorageContextUtils.java]: @shiyiyue1102 Please review again， thank you.

File: config/src/main/java/com/alibaba/nacos/config/server/service/repository/embedded/BaseDatabaseOperate.java [hunk 3]
```diff
@@ -234,7 +239,7 @@ default Boolean update(TransactionTemplate transactionTemplate, JdbcTemplate jdb
             } catch (CannotGetJdbcConnectionException e) {
                 FATAL_LOG.error("[db-error] sql : {}, args : {}, error : {}", errSql[0], args[0], e.toString());
                 throw e;
-            } catch (DataAccessException e) {
+            } catch (DataAccessException | IllegalTransactionStateException e) {
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/service/repository/embedded/BaseDatabaseOperate.java:242]: illegal transaction exception should not be logged as DataAccessException,besides ,it is not a fatal exception  and should not be log into FATAL_LOG

Comment (karsonto) [reply config/src/main/java/com/alibaba/nacos/config/server/service/repository/embedded/BaseDatabaseOperate.java]: Modified according to your suggestion
 @shiyiyue1102 please review again.

File: config/src/main/java/com/alibaba/nacos/config/server/service/sql/ModifyRequest.java [hunk 4]
```diff
@@ -33,6 +33,8 @@ public class ModifyRequest implements Serializable {
     
     private String sql;
     
+    private boolean rollBackOnUpdateFail = Boolean.TRUE;
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/service/sql/ModifyRequest.java:36]: default value should be “false” to compatible the former

Comment (shiyiyue1102) [reply config/src/main/java/com/alibaba/nacos/config/server/service/sql/ModifyRequest.java:36]: And only in config cas update method ,this value should be true,or you should test all scenes that affected by this modification

**Comment by shiyiyue1102** [comment]
Integration Test fails, fix it and I will approve this pr.

**Review by shiyiyue1102 — APPROVED** [review]

**Pull request merged**